src/main/java/com/mwt/explorers/SoftmaxExplorer.java

```diff
@@ -46,43 +46,45 @@ public DecisionTuple chooseAction(long saltedSeed, T context) {
     PRG random = new PRG(saltedSeed);
 
     // Invoke the default scorer function
-    ArrayList<Float> scores = defaultScorer.scoreActions(context);
+    List<Float> scores = defaultScorer.scoreActions(context);
     int numScores = scores.size();
     if (numScores != getNumActions(context)) {
       throw new RuntimeException("The number of scores returned by the scorer must equal number of actions");
     }
 
     int actionIndex = 0;
-    float actionProbability = 1.0f;
-    Float maxScore = Float.MIN_VALUE;
-    for (int i = 0; i < numScores; i++) {
-      if (maxScore < scores.get(i)) {
-        maxScore = scores.get(i);
+    float actionProbability = 1f;
+    float maxScore = Float.MIN_VALUE;
```

Comment:
Changed to primitive `float` because it is compared current `score` so we don't want to keep doing unnecessary unboxing in the loop.